marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/MarathonExtension.kt

```diff
@@ -7,183 +7,120 @@ import com.malinskiy.marathon.android.configuration.FileSyncConfiguration
 import com.malinskiy.marathon.android.configuration.SerialStrategy
 import com.malinskiy.marathon.android.configuration.TimeoutConfiguration
 import com.malinskiy.marathon.execution.policy.ScreenRecordingPolicy
-import groovy.lang.Closure
-import org.gradle.api.Project
+import org.gradle.api.Action
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.Property
+import org.gradle.kotlin.dsl.mapProperty
+import org.gradle.kotlin.dsl.property
+import java.io.Serializable
+import javax.inject.Inject
 
-open class MarathonExtension(project: Project) {
-    var name: String = "Marathon"
+open class MarathonExtension @Inject constructor(objects: ObjectFactory) : Serializable {
```

Comment:
There is one more trick
If you make a class `abstract` and mark property `abstract` Gradle will initialize them for you
It helps to reduce the amount of code